packages/cli/src/tools/config/index.js

```diff
@@ -2,36 +2,90 @@
  * @flow
  */
 import path from 'path';
-import {mapValues} from 'lodash';
 import chalk from 'chalk';
-
+import {logger, inlineString} from '@react-native-community/cli-tools';
+import * as ios from '@react-native-community/cli-platform-ios';
+import * as android from '@react-native-community/cli-platform-android';
 import findDependencies from './findDependencies';
 import resolveReactNativePath from './resolveReactNativePath';
 import findAssets from './findAssets';
-import makeHook from './makeHook';
 import {
   readConfigFromDisk,
   readDependencyConfigFromDisk,
 } from './readConfigFromDisk';
-
 import {type ConfigT} from 'types';
-
 import assign from '../assign';
 import merge from '../merge';
 import resolveNodeModuleDir from './resolveNodeModuleDir';
-/**
- * Built-in platforms
- */
-import * as ios from '@react-native-community/cli-platform-ios';
-import * as android from '@react-native-community/cli-platform-android';
-import {logger, inlineString} from '@react-native-community/cli-tools';
+
+function getDependencyConfig(
+  root,
+  dependencyName,
+  finalConfig,
+  config,
+  userConfig,
+  isPlatform,
+) {
+  return merge(
+    {
+      root,
+      name: dependencyName,
+      platforms: Object.keys(finalConfig.platforms).reduce(
+        (dependency, platform) => {
+          // Linking platforms is not supported
+          dependency[platform] = isPlatform
+            ? null
+            : finalConfig.platforms[platform].dependencyConfig(
+                root,
+                config.dependency.platforms[platform] || {},
+              );
+          return dependency;
+        },
+        {},
+      ),
+      assets: findAssets(root, config.dependency.assets),
+      hooks: config.dependency.hooks,
```

Comment:
the only change in this moved block is this line – we pass the `hooks` string directly, without mapping to functions (which are not serializable, hence not displayed after JSON.stringify runs on config)

Reply:
Yeah, right!

Reply:
I guess an alternative take would be to provide a special `toString` implementation as a result of `makeHook` call to make it serializable?

Reply:
I think it's better to keep the whole config serializable anyway, less things to remember

Reply:
Yeah, I was suggesting to make `makeHook` return a serializable function (`toString` returns a path). 

They are going to be gone (most likely) in the future anyway, so not bothering to much with that one.